bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationHandler.java

```diff
@@ -0,0 +1,410 @@
+                    valueConfiguration = command.toString();
+                    String valueConfigurationFromTransformed = String.valueOf(stringToSeconds(valueConfiguration));
+                    dxsEntriesConf = "33556239";
+                    preSetExecuteConfigurationChanges(httpClient, url, username, password, dxsEntriesConf,
+                            valueConfigurationFromTransformed);
+                    break;
+                case SecondGenerationBindingConstants.CHANNEL_BATTERYCHARGETIMETOSET:
+                    valueConfiguration = command.toString();
+                    String valueConfigurationToTransformed = String.valueOf(stringToSeconds(valueConfiguration));
+                    dxsEntriesConf = "33556240";
+                    preSetExecuteConfigurationChanges(httpClient, url, username, password, dxsEntriesConf,
+                            valueConfigurationToTransformed);
+                    break;
+                case SecondGenerationBindingConstants.CHANNEL_MAXDEPTHOFDISCHARGESET:
+                    valueConfiguration = command.toString();
+                    dxsEntriesConf = "33556247";
+                    preSetExecuteConfigurationChanges(httpClient, url, username, password, dxsEntriesConf,
+                            valueConfiguration);
+                    break;
+                case SecondGenerationBindingConstants.CHANNEL_SHADOWMANAGEMENTSET:
+                    valueConfiguration = command.toString();
+                    dxsEntriesConf = "33556483";
+                    preSetExecuteConfigurationChanges(httpClient, url, username, password, dxsEntriesConf,
+                            valueConfiguration);
+                    break;
+                case SecondGenerationBindingConstants.CHANNEL_EXTERNALMODULECONTROLSET:
+                    valueConfiguration = command.toString();
+                    dxsEntriesConf = "33556482";
+                    preSetExecuteConfigurationChanges(httpClient, url, username, password, dxsEntriesConf,
+                            valueConfiguration);
+                    break;
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        // Set channel configuration parameters
+        channelConfigs = SecondGenerationChannelConfiguration.getChannelConfiguration();
+        channelConfigsExt = SecondGenerationChannelConfiguration.getChannelConfigurationExt();
+        channelConfigsExtExt = SecondGenerationChannelConfiguration.getChannelConfigurationExtExt();
+        channelConfigsConfigurable = SecondGenerationChannelConfiguration.getChannelConfigurationConfigurable();
+
+        // Set inverter configuration parameters
+        final SecondGenerationInverterConfig inverterConfig = getConfigAs(SecondGenerationInverterConfig.class);
+        this.inverterConfig = inverterConfig;
+
+        // Initialize Gson object gson
+        final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        this.gson = gson;
```

Comment:
You could initialize this in the constructor to get rid of the compiler warning about null values. You don't see them currently as you supressed them in `refresh()`.